base/dict.jl

```diff
@@ -270,7 +270,11 @@ function rehash!(h::Dict{K,V}, newsz = length(h.keys)) where V where K
     return h
 end
 
-function sizehint!(d::Dict, newsz)
+max_values(::Type) = typemax(Int)
```

Comment:
I feel uncomfortable with this one (in particular on 32 bit machines): `sizehint!` doesn't seem to make any garantee, but for `union!(s::Set, xs)`, could it be that `s` could contain more than `typemax(Int)` elements?